clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -566,8 +573,11 @@ public CompletableFuture<Void> acknowledgeOnClose(final Map<TopicIdPartition, Ac
                 for (TopicIdPartition tip : sessionHandler.sessionPartitions()) {
                     Acknowledgements acknowledgements = acknowledgementsMap.getOrDefault(tip, Acknowledgements.empty());
 
-                    if (fetchAcknowledgementsMap.get(tip) != null) {
-                        acknowledgements.merge(fetchAcknowledgementsMap.remove(tip));
+                    Acknowledgements acksFromShareFetch = fetchAcknowledgementsToSend.remove(tip);
+
+                    if (acksFromShareFetch != null) {
+                        acknowledgements.merge(acksFromShareFetch);
+                        fetchAcknowledgementsInFlight.put(tip, acksFromShareFetch);
```

Comment:
It doesn't seem necessary to put the acknowledgements into the `fetchAcknowledgementsInFlight` map in this case. We are closing the share session using a `ShareAcknowledge` and this code is just picking up any acknowledgements which were intended to be piggybacked on the next fetch. They have essentially been taken over by this logic and don't need to be accounted for as fetch acknowledgments in flight.

Reply:
Yeah right makes sense, these acks would just remain over in the inFlight map. I have now removed the line and updated the PR. Thanks.